app/controllers/manage_competencies/base_controller.rb

```diff
@@ -0,0 +1,23 @@
+module ManageCompetencies
+  class BaseController < ApplicationController
+    layout 'manage_competencies'
+
+    before_action :authenticate_user!
+    before_action :require_supervisor!
+    before_action :set_security_headers
+
+    # Scope for I18n locale, used by _text helpers.
+    def text_namespace
+      :manage_competencies
+    end
+
+    private
+
+    def require_supervisor!
+      return if current_user.role == Types::SUPERVISOR_ROLE
+
+      render status: :not_found, template: 'errors/not_found', layout: 'external'
```

Comment:
Doing this:
```
      raise ForbiddenError, 'Must be a supervisor'
```
will have the same effect

Reply:
Thanks, made that change